evaluator/src/main/kotlin/PackageRule.kt

```diff
@@ -70,6 +70,27 @@ open class PackageRule(
         licenseRules.forEach { it.evaluate() }
     }
 
+    /**
+     * A [RuleMatcher] that checks whether a severity of any vulnerability of the [package][pkg] using the specified
+     * [scoringSystem] is higher than [maxValue]. Checked using the [severityComparator]. Returns false if the severity
+     * is equal to [maxValue].
+     */
+    fun hasVulnerability(maxValue: String, scoringSystem: String, severityComparator: Comparator<String>) =
+        object : RuleMatcher {
+            override val description =
+                "Higher severity than $maxValue defined by $scoringSystem found for ${pkg.id.toCoordinates()}."
```

Comment:
Note that the `description` is not actually user-facing, but only used in logs currently. So it might make more sense to use something shorter here; see the descriptions of other rules.